src/uucp_break.ml

```diff
@@ -35,6 +35,38 @@ let sentence u = Array.unsafe_get Low.sentence_of_int (Low.sentence u)
 
 let east_asian_width u = Uucp_rmap.get Uucp_break_data.east_asian_width_map u
 
+let err_ctrl_char e =
+  invalid_arg (Printf.sprintf "0x%02X is a control character" e)
+
+let err_not_uchar e =
+  invalid_arg (Printf.sprintf "U+%04X is not a unicode scalar value." e)
+
+let terminal_width_hint =
+  let gc = Uucp_gc.general_category in
+  function
+  (* C0 (without 0x00) + DELETE + C1 is an error. *)
+  |u when 0 < u && u <= 0x1f || 0x7f <= u && u <= 0x9f -> err_ctrl_char u
```

Comment:
Problem is, `0` in those cases is more wrong than usual. If you miss one cursor position to the right, stuff gets mis-aligned, but if you emit a `'\r'`, it all goes to hell.

I want the function to classify those completely pathological cases. There is some knowledge embedded in there: `0x00` is OK, but `0x7f` (which is neither **C0** nor **C1**) is a problem, etc.

So it could return `-1`, `666`, or `1234321`, to denote that it now *knows* that the measurement won't work. Or it could be lifted to `int option` and return `None`. It's just that an exception + precondition is cheaper to the client sanitizing its input, both in syntax and execution.

Or, stated differently, we have advance knowledge that some of the unicode scalar values have no meaningful terminal width at all. Pick your way of signaling that fact.